palantir-java-format/build.gradle

```diff
@@ -30,40 +30,28 @@ dependencies {
     annotationProcessor 'org.derive4j:derive4j'
 }
 
-def exports = [
-        'jdk.compiler/com.sun.tools.javac.file',
-        'jdk.compiler/com.sun.tools.javac.main',
-        'jdk.compiler/com.sun.tools.javac.parser',
-        'jdk.compiler/com.sun.tools.javac.tree',
-        'jdk.compiler/com.sun.tools.javac.util',
-        'jdk.compiler/com.sun.tools.javac.code',
-        'jdk.compiler/com.sun.tools.javac.api'
-]
-
-def jvmArgList = exports.collect { value -> "--add-exports=${value}=ALL-UNNAMED".toString() }
-
-tasks.withType(JavaCompile).configureEach {
-    options.errorprone.disable 'StrictUnusedVariable'
-
-    // Allow access to internal javac apis
-    options.compilerArgs += jvmArgList
-
-    if (JavaVersion.current() < JavaVersion.VERSION_14) {
-        excludes = ['**/Java14InputAstVisitor.java']
-    }
+moduleJvmArgs {
+    exports().addAll(
+            'jdk.compiler/com.sun.tools.javac.file',
+            'jdk.compiler/com.sun.tools.javac.main',
+            'jdk.compiler/com.sun.tools.javac.parser',
+            'jdk.compiler/com.sun.tools.javac.tree',
+            'jdk.compiler/com.sun.tools.javac.util',
+            'jdk.compiler/com.sun.tools.javac.code',
+            'jdk.compiler/com.sun.tools.javac.api')
 }
 
-tasks.withType(Test).configureEach {
-    jvmArgs = jvmArgList
+tasks.withType(JavaCompile).named('compileJava') {
+    // By setting the source and target compatibility to 11, while still using a higher level
+    // JDK for compilation, we can compile against the AST classes we need to support >11 source
+    // features but still run on 11 JDKs (with the addition of checking the java version at runtime
+    // and dynamically loading the correct formatter implementation)
+    sourceCompatibility = JavaVersion.VERSION_11
+    targetCompatibility = JavaVersion.VERSION_11
 }
 
 tasks.withType(Javadoc).configureEach {
-    // Allow access to internal javac apis
-    options.optionFiles << file('../gradle/javadoc.options')
-
-    if (JavaVersion.current() < JavaVersion.VERSION_14) {
-        exclude '**/Java14InputAstVisitor.java'
-    }
+    enabled = false
```

Comment:
We rarely if ever use javadoc - I'll need to check this doesn't break publishing